### ISSUE TITLE
Remove support for string exceptions from twisted.python.failure

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5948]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#5948|
|Type|enhancement|
|Created|2012-09-06 15:03:32Z|
|Branch|https://github.com/twisted/twisted/tree/no-string-failures-5948|

Python 2.6 is now minimal version, we don't need support for string exceptions and it's making Python 3 port harder.

<details><summary>Searchable metadata</summary>

```
trac-id__5948 5948
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_no_string_failures_5948 branches/no-string-failures-5948
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1346943812000000 1346943812000000
changetime__1346955947000000 1346955947000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [35530]) Branching to 'no-string-failures-5948'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst set owner to @exarkun|
|-|-|

Ready for review:
buildbot.twistedmatrix.com/boxes-supported?branch=/branches/no-string-failures-5948

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @itamarst|
|-|-|

  1. I'm mildly concerned about `twisted.spread.pb.CopiedFailure`, but as far as I can _tell_, it overrides the only `Failure` method which is losing support for a string `type` attribute (`printTraceback`), so I guess I shouldn't worry (at least not any more than I already did about the mere existence of a `Failure` subclass in `twisted.spread`).
  1. There's a new pyflakes warning about `test_failure.py` (_'util' imported but unused_).

Otherwise looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set status to `closed`|
|-|-|

(In [35542]) Merge no-string-failures-5948: no more Failure support for string exceptions.

Author: itamar
Review: exarkun
Fixes: [#5948](https://github.com/twisted/twisted/issues/5948)

Failure will no longer even pretend to work with string exceptions.
